File: Top 100 Codes/01-Getting Started/23-Perfect number/Perfectnumber.cpp
#include <iostream>
using namespace std;

int main() {
    // Declare variables
    int n, sum = 0;

    // Prompt the user to enter a number
    cout << "Enter a number to check if it is a perfect number: ";
    cin >> n;

    // Calculate the sum of divisors of the number (excluding itself)
    for (int i = 1; i < n; i++) {
        if (n % i == 0) { // Check if 'i' is a divisor of 'n'
            sum += i;    // Add the divisor to the sum
        }
    }

    // Check if the sum of divisors equals the number itself
    if (sum == n) {
        cout << n << " is a perfect number." << endl;
    } else {
        cout << n << " is not a perfect number." << endl;
    }

    // Time complexity: O(N) 
    // Space complexity: O(1)

    return 0;
}

File: Top 100 Codes/01-Getting Started/23-Perfect number/Perfectnumber.py
# Prompt the user to enter a number
n = int(input("Enter a number to check if it is a Perfect number: "))
sum_of_divisors = 0  # Initialize the sum of divisors to 0

# Loop through numbers from 1 to n-1 to find all divisors
for i in range(1, n):
    if n % i == 0:  # Check if 'i' is a divisor of 'n'
        sum_of_divisors += i  # Add the divisor to the sum

# Check if the sum of divisors equals the number itself
if sum_of_divisors == n:
    print(f"{n} is a Perfect number.")
else:
    print(f"{n} is not a Perfect number.")
